src/software/jetson_nano/rotary_encoder/RotaryEncoder.py

```diff
@@ -0,0 +1,123 @@
+import Jetson.GPIO as GPIO
+
+# Channel B and Channel A states
+STATE_0 = [0, 0]
+STATE_1 = [0, 1]
+STATE_2 = [1, 1]
+STATE_3 = [1, 0]
+
+CLOCKWISE = 1
+COUNTERCLOCKWISE = -1
+IDLE = 0
+
+
+class RotaryEncoder:
+    def __init__(
+        self,
+        A_PIN,
+        B_PIN,
+        BUTTON_PIN,
+        on_clockwise_rotate,
+        on_counterclockwise_rotate,
+        on_click,
+    ):
+        self.A_PIN = A_PIN
+        self.B_PIN = B_PIN
+        self.BUTTON_PIN = BUTTON_PIN
+
+        self.on_clockwise_rotate = on_clockwise_rotate
+        self.on_counterclockwise_rotate = on_counterclockwise_rotate
+        self.on_click = on_click
+
+        self.counts_per_rotation = 4
+
+    def init(self):
```

Comment:
nit: maybe call this setup? so its not super close to __init__

Reply:
Changed the function name to `setup`